packages/data/src/world/mq/jabu_jabu_mq.yml

```diff
@@ -14,64 +14,180 @@
+    "Jabu-Jabu Ruto Section Bottom Floor": "true"
+    "Jabu-Jabu Main Elevator Room Lower": "event(JABU_MQ_WATER_SPOUTS)"
+  events:
+    JABU_MQ_WATER_SPOUTS: "soul_lizalfos_dinalfos && soul_like_like && soul_enemy(SOUL_ENEMY_STINGER) && has_weapon"
+  locations:
+    "MQ Jabu-Jabu Boomerang Chest": "event(JABU_MQ_WATER_SPOUTS)"
+    "MQ Jabu-Jabu SoT Room Lower Chest": "true"
+    "MQ Jabu-Jabu Pot Boomerang Room 1": "true"
+    "MQ Jabu-Jabu Pot Boomerang Room 2": "true"
+    "MQ Jabu-Jabu GS SoT Block": "gs && can_play_time && can_damage_skull"
+    "MQ Jabu-Jabu Grass Boomerang Room": "can_cut_grass"
+"Jabu-Jabu Basement Ledge Before Big Octo":
+  dungeon: JJ
+  exits:
+    "Jabu-Jabu Ruto Section Bottom Floor": "true"
+    "Jabu-Jabu Basement Ledge Before Big Octo After Green Tentacle": "event(JABU_TENTACLE_GREEN)" #This tentacle can block in door rando, so it needs its own region.
+  locations:
     "MQ Jabu-Jabu Wonder Item Basement Right Cow 1": "can_use_slingshot"
     "MQ Jabu-Jabu Wonder Item Basement Right Cow 2": "can_use_slingshot"
     "MQ Jabu-Jabu Wonder Item Basement Right Cow 3": "can_use_slingshot"
     "MQ Jabu-Jabu Wonder Item Basement Left Cow 1": "can_use_slingshot"
     "MQ Jabu-Jabu Wonder Item Basement Left Cow 2": "can_use_slingshot"
     "MQ Jabu-Jabu Wonder Item Basement Left Cow 3": "can_use_slingshot"
-    "MQ Jabu-Jabu Wonder Item After Big Octo": "event(JABU_BIG_OCTO) && can_use_slingshot"
-    "MQ Jabu-Jabu Wonder Item Platforms Cow": "event(JABU_BIG_OCTO) && can_use_slingshot"
-"Jabu-Jabu Back":
+"Jabu-Jabu Basement Ledge Before Big Octo After Green Tentacle":
+  dungeon: JJ
+  exits:
+    "Jabu-Jabu Basement Ledge Before Big Octo": "event(JABU_TENTACLE_GREEN)" #This tentacle can block in door rando, so it needs its own region.
+    "Jabu-Jabu Big Octo Room": "true"
+"Jabu-Jabu Big Octo Room":
   dungeon: JJ
   exits:
-    "Jabu-Jabu Main": "true"
+    "Jabu-Jabu Basement Ledge Before Big Octo After Green Tentacle": "true"
+    "Jabu-Jabu Above Big Octo Fight": "event(JABU_BIG_OCTO)"
   events:
-    "JABU_TENTACLE_BLUE": "((can_use_sticks && soul_like_like && can_use_slingshot) || has_fire) && can_boomerang && soul_enemy(SOUL_ENEMY_PARASITE)"
-    "JABU_TENTACLE_RED": "can_boomerang && soul_enemy(SOUL_ENEMY_PARASITE)"
-    "JABU_TENTACLE_GREEN": "event(JABU_TENTACLE_BLUE)"
+    JABU_BIG_OCTO: "event(JABU_TENTACLE_GREEN) && soul_octorok && (can_use_sticks || has_weapon) && soul_ruto"
+"Jabu-Jabu Above Big Octo Fight":
+  dungeon: JJ
+  exits:
+    "Jabu-Jabu Room After Above Big Octo": "can_use_slingshot"
+  locations:
+    "MQ Jabu-Jabu Grass Big Octo Top 1": "can_cut_grass"
+    "MQ Jabu-Jabu Grass Big Octo Top 2": "can_cut_grass"
+    "MQ Jabu-Jabu Wonder Item After Big Octo": "can_use_slingshot"
+"Jabu-Jabu Room After Above Big Octo":
+  dungeon: JJ
+  exits:
+    "Jabu-Jabu Above Big Octo Fight": "true"
+    "Jabu-Jabu Main Elevator Room High Ledge": "(climb_anywhere && can_play_elegy) || can_boomerang"
```

Comment:
Should be able to do this with Elegy and Boomerang too. You need slingshot for the cow to spawn the crates to take to the door.

Reply:
Hookshot through the flesh seems very glitchy to me. I don't think we should expect that.

Otherwise, fixed. Added an event too as I noticed the crates PR for this section was wrong too anyway,